includes/rest-api/class-amp-rest-stories-controller.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * Class AMP_REST_Stories_Controller
+ *
+ * @package AMP
+ */
+
+/**
+ * Override the WP_REST_Posts_Controller class to add `post_content_filtered` to REST request.
+ *
+ * Class AMP_REST_Stories_Controller
+ */
+class AMP_REST_Stories_Controller extends WP_REST_Posts_Controller {
+	/**
+	 * Prepares a single story for create or update. Add post_content_filtered field to save/insert.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return stdClass|WP_Error Post object or WP_Error.
+	 */
+	protected function prepare_item_for_database( $request ) {
+		$prepared_story = parent::prepare_item_for_database( $request );
+
+		if ( is_wp_error( $prepared_story ) ) {
+			return $prepared_story;
+		}
+
+		if ( isset( $request['content_filtered'] ) ) {
+			$prepared_story->post_content_filtered = $request['content_filtered'];
+			if ( isset( $request['content_filtered']['raw'] ) ) {
+				$prepared_story->post_content_filtered = $request['content_filtered']['raw'];
+			}
+
+			$prepared_story->post_content_filtered = wp_json_encode( $prepared_story->post_content_filtered );
+		}
+
+		return $prepared_story;
+	}
+
+	/**
+	 * Prepares a single story output for response. Add post_content_filtered field to output.
+	 *
+	 * @param WP_Post         $post Post object.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $post, $request ) {
+		$response = parent::prepare_item_for_response( $post, $request );
+		$fields   = $this->get_fields_for_response( $request );
+		$data     = $response->get_data();
+		$schema   = $this->get_item_schema();
+
+		if ( in_array( 'content_filtered', $fields, true ) ) {
+			$post_content_filtered    = json_decode( $post->post_content_filtered );
```

Comment:
This needs to ensure an array is returned rather than an object, right?

```suggestion
			$post_content_filtered    = json_decode( $post->post_content_filtered, true );
```